File: Rob_Jaap/java/src/main/java/com/chariotit/uva/sc/qdsl/ast/qls/node/Stylesheet.java
package com.chariotit.uva.sc.qdsl.ast.qls.node;

import com.chariotit.uva.sc.qdsl.ast.common.SourceFilePosition;
import com.chariotit.uva.sc.qdsl.ast.qls.visitor.NodeVisitor;

import java.util.List;

public class Stylesheet extends AstNode {

    private List<Page> pages;
    private String label;

    public Stylesheet(List<Page> pages, String label, SourceFilePosition filePosition) {
        super(filePosition);
        this.pages = pages;
        this.label = label;
    }

    public List<Page> getPages() {
        return pages;
    }

    public void setPages(List<Page> pages) {
        this.pages = pages;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    @Override
    public void acceptVisitor(NodeVisitor visitor) {
        for (Page page : pages) {
            page.acceptVisitor(visitor);
        }

        visitor.visitStylesheet(this);
    }
}

File: nitpickers/pyql/pyqls/ast/nodes/stylesheet.py
from util.ast import ASTNode


class StyleSheet(ASTNode):

    def __init__(self, location, name, block):
        super().__init__(location)
        self._name = name
        self._block = block

    @property
    def name(self):
        return self._name

    @property
    def block(self):
        return self._block

    def __repr__(self):
        return "StyleSheet {0}".format(self.name)

    __str__ = __repr__
